### DISCUSSION TITLE
Import predefined activities for users

### DISCUSSION THREAD

**Discussion by Dany1410** [kimai/kimai#3420] [Questions & Answers]
Hello everbody and Kevin,

I imported the data from Kimai Version 1 to Version 2. After that and checking the application I noticed the following.
When a user is permitted to select a project for time recording and wants to select a activity, all activites for this project are displayed. In Kimai Version 1 there was a predefined selection of activities the admin has chosen for a user to use. I know this feature is still there. But is it possible to migrate this information as well or do I have to adjust this afterwards manually?

I had a look at your possible extensions for the import command but nothing really seems to fit my need and I did not use the flag --skip-team-activities.
Can you please help me?

Best, 
Dany1410

**Comment by kevinpapst** [comment]
Didn't we talk about that issue already in your other thread?
Kimai 2 is not working as Kimai 1. If you need the exact behavior of Kimai 1 you have to use Kimai 1.
If you use --global flag during import, all activities will be become available for all projects.

**Reply by Dany1410** [reply]
Well, yes we did - kind of. Leaving the --global flag behind worked for seeing all related activities for a specific project. So far so good.
But I want the activities to be imported with the predefined permission per user. For instance, there is project 1 and 10 related activities to it. The Admin defined that a specific user is allowed to select/see/use only 2 of them when he is recording his time (as it is in Kimai1). Is it possible to migrate this information or do I have to adjust this manually after migrating the data? :)

**Reply by kevinpapst** [reply]
I have never seen or used that features in Kimai 1.
And there is no such feature in Kimai 2. 
You would have to define a Team for every user and assign these team permissions to the imported activities ... 
Sounds way too complicated for a very limited use-case.

**Reply by Dany1410** [reply]
Correct, that would be the manual way to do it. I was just wondering if this information is importable.
Unfortunately, we need this complicated view to reduce false time records from users since we have quite a lot activities per project.
Anyway, thank you for your help!

**Reply by kevinpapst** [reply]
If I remember correct, you have quite a lot of data and doing that manually sounds like a task that can take forever.
Maybe it is simpler to adjust the importer, so it is done automatically for you.

**Reply by Dany1410** [reply]
That would be awesome. But I am not familiar with the importer command. I would need to have a look at it and work in.
Is this command based on php?
In contrast, we have 20+ active users. I would either adjust it in the database or directly in the kimai application. There is already a Team per user. I would just need to asign the specific activities to those teams.

**Reply by kevinpapst** [reply]
Yes and I am not proud of the code 😁 but it does its job:
https://github.com/kevinpapst/kimai2/blob/master/src/Command/KimaiImporterCommand.php

**Reply by Dany1410** [reply]
Okay, I have absolutely no idea how to adjust the code to automate my need....
